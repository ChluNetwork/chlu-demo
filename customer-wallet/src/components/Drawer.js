import React from 'react'
import PropTypes from 'prop-types'
// components
import Drawer from 'material-ui/Drawer'
import MenuItem from 'material-ui/MenuItem'

const DrawerComponent = ({ toggleDrawer, modalOpen }) => (
  <Drawer
    onRequestChange={toggleDrawer}
    open={modalOpen}
    docked={false}>
    <MenuItem>Checkout</MenuItem>
    <MenuItem>Customer Wallet</MenuItem>
    <MenuItem>Vendor Profile</MenuItem>
  </Drawer>
)

DrawerComponent.propTypes = {
  toggleDrawer: PropTypes.func.isRequired,
  modalOpen: PropTypes.bool
}

export default DrawerComponent
