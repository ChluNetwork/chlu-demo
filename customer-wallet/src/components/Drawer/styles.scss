@import 'styles/constants.scss';

.drawer-link {
  color: $mainColor;
  text-decoration: none;

  &-active {
    >div {
      background: rgb(229,229,229);
    }
  }
}
