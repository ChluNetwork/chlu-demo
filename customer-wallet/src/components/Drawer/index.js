import React from 'react'
import PropTypes from 'prop-types'
// redux
import { connect } from 'react-redux'
// components
import { Link } from 'react-router'
import Drawer from 'material-ui/Drawer'
import MenuItem from 'material-ui/MenuItem'
// styles
import './styles.css'
// data
import { linksForCustomer, linksForVendor, linksForDemonstrator } from 'shared-libraries/lib/fixtures/links'

const style = {
  color: 'inherit',
  background: 'inherit'
}

const DrawerComponent = ({ toggleDrawer, drawerOpen, userType }) => {
  let links = []

  switch(userType){
    case 'customer': links = linksForCustomer
      break
    case 'vendor': links = linksForVendor
      break
    case 'demonstrator': links = linksForDemonstrator
      break
    default: links = []
      break
  }

  return (
    <Drawer
      onRequestChange={toggleDrawer}
      open={drawerOpen}
      docked={false}>
      {links.map(({ label, href }, idx) => (
        <Link
          to={href}
          className='drawer-link'
          activeClassName='drawer-link-active'
          key={idx}
          onClick={toggleDrawer}
        >
          <MenuItem style={style} >
            {label}
          </MenuItem>
        </Link>
      ))}
    </Drawer>
  )
}

DrawerComponent.propTypes = {
  toggleDrawer: PropTypes.func.isRequired,
  drawerOpen: PropTypes.bool,
  userType: PropTypes.string.isRequired
}

const mapStateToProps = state => ({
  userType: state.data.profile.data ? state.data.profile.data.userType : ''
})

export default connect(mapStateToProps)(DrawerComponent)
