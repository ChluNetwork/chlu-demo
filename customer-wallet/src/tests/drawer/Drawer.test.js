import React from 'react'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import toJson from 'enzyme-to-json'
import Drawer from 'components/Drawer'

const mockStore = configureStore([])
const store = mockStore({ open: false })

describe('<Drawer />', () => {

  it('should render correctly drawer component', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Drawer
          toggleDrawer={() => null}
          drawerOpen
        />
      </Provider>
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
