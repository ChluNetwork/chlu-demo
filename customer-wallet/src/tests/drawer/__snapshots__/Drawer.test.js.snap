// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Drawer /> should render correctly drawer component 1`] = `
<Drawer
  disableSwipeToOpen={false}
  docked={false}
  onRequestChange={[Function]}
  open={true}
  openSecondary={false}
  swipeAreaWidth={30}
  width={null}
  zDepth={2}
>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  >
    <Link
      className="drawer-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/checkout"
    >
      Checkout
    </Link>
  </MenuItem>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  >
    <Link
      className="drawer-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/"
    >
      Customer Wallet
    </Link>
  </MenuItem>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  >
    <Link
      className="drawer-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/profile"
    >
      Vendor Profile
    </Link>
  </MenuItem>
  <MenuItem
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "top",
      }
    }
    checked={false}
    desktop={false}
    disabled={false}
    focusState="none"
    insetChildren={false}
    targetOrigin={
      Object {
        "horizontal": "left",
        "vertical": "top",
      }
    }
  >
    <Link
      className="drawer-link"
      onlyActiveOnIndex={false}
      style={Object {}}
      to="/vendorWallet"
    >
      Vendor Wallet
    </Link>
  </MenuItem>
</Drawer>
`;
